api/protogen/protogen.go

```diff
@@ -270,12 +273,55 @@ func terraformAttributeTypeToProtoType(attrName string, attrType attr.Type) (rep
 		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
 	case types.SetType:
 		return terraformRepeatedAttributeTypeToProtoType(attrName, v.ElementType())
-	// TODO: Add support for nested objects.
+	case types.ObjectType:
+		return terraformObjectAttributeTypeToProtoType(attrName, v)
+
 	default:
 		return false, "", nil, fmt.Errorf("unsupported Terraform type: %s", attrType.String())
 	}
 }
 
+// terraformObjectAttributeTypeToProtoType converts a Terraform object attribute to a Protocol Buffer message type.
+func terraformObjectAttributeTypeToProtoType(attrName string, obj types.ObjectType) (repeated bool, protoType string, messages *message, err error) {
```

Comment:
```suggestion
func terraformObjectAttributeTypeToProtoType(attrName string, obj types.ObjectType) (repeated bool, protoType string, msg *message, err error) {
```
Or something like that (not plural).
Please fix the same in `terraformAttributeTypeToProtoType`. Thanks!

Reply:
fixed